src/app/routes/routes.jsx

```diff
@@ -1,25 +1,32 @@
 // React library
 import React from 'react';
 // Import Router
-import { IndexRoute, Route, Redirect } from 'react-router';
+import { IndexRoute, Router, Route, Redirect } from 'react-router';
 
 // Import components
 import App from '../components/Application/Application';
 import Main from '../components/Application/Main';
 import Error404Page from '../components/Error404Page/Error404Page';
 
-import appConfig from '../../../appConfig';
+import config from '../../../appConfig';
 
-// Routes we need
+// Routes are specified directly here in order to properly match and route to
+// the 404 page appropriately.
 const routes = {
   client: (
-    <Route path={`${appConfig.baseUrl}`} component={App}>
-      <IndexRoute component={Main} />
-      <Route path={`${appConfig.baseUrl}404`} component={Error404Page} />
-      <Route path={`${appConfig.baseUrl}:type`} component={Main} />
-      <Route path={`${appConfig.baseUrl}:type/:time`} component={Main} />
-      <Redirect from="*" to={`${appConfig.baseUrl}404`} />
-    </Route>
+    <Router>
+      <Route path={`${config.baseUrl}`} component={App}>
+        <IndexRoute component={Main} />
+        <Route path={`${config.base404}`} status={404} component={Error404Page} />
+        <Route path={`${config.baseUrl}staff-picks`} component={Main} />
```

Comment:
This doesn't work if you remove lines 21-23 and make the `:time` param optional?